ui/js/models/resultsets_store.js

```diff
@@ -289,19 +289,40 @@ treeherder.factory('ThResultSetStore', [
         };
 
         var addRunnableJobs = function(repoName, resultSet) {
-            return ThRunnableJobModel.get_list(repoName).then(function(jobList) {
-                var id = resultSet.id;
-                _.each(jobList, function(job) {
-                    job.result_set_id = id;
-                    job.id = thAggregateIds.escape(job.result_set_id + job.ref_data_name);
-                });
-
-                if (jobList.length === 0) {
-                    resultSet.isRunnableVisible = false;
-                    thNotify.send("No new jobs available");
+            // first check whether the Gecko Decision Task has completed
+            var tcURLPromise;
+            var platform = _.find(resultSet.platforms, {
+                "name": "gecko-decision",
+                "groups": [{"jobs": [{"state": "completed"}]}]});
+            if (platform) {
+                // Gecko Decision Task has been completed.
+                // Let's fetch the URL of all_tasks.json
+                var job_guid = platform.groups[0].jobs[0].job_guid;
+                tcURLPromise = ThJobDetailModel.getJobDetails(job_guid, {timeout: null});
+            }
+            if (!tcURLPromise) {
+                tcURLPromise = Promise.resolve(false);
+            }
+            return tcURLPromise.then(function(results) {
+                var taskID = "";
+                if (results) {
```

Comment:
Would you add some commends in the various states of the if..thens and such?  It would help me reviewing now, and when I need to look at it in some future time to help me remember.  :)